barretenberg/cpp/src/barretenberg/proof_system/circuit_builder/goblin_ultra_circuit_builder.cpp

```diff
@@ -254,6 +222,44 @@ template <typename FF> void GoblinUltraCircuitBuilder_<FF>::set_goblin_ecc_op_co
     equality_op_idx = this->put_constant_variable(FF(EccOpCode::EQUALITY));
 }
 
+/**
+ * @brief Create a calldata lookup/read gate
+ *
+ * @tparam FF
+ * @param read_index Raw index into calldata array
```

Comment:
The way it's implemented there is no connection with other parts of the circuit, so effectively we're saying "We're reading something". A malicious prover can read from any slot then

Reply:
Yeah it's a fair point. I'm kind of being lazy here and leaving things in an incomplete state. Are you suggesting that the index should instead be an index into the variables array?